sched/init/nx_start.c

```diff
@@ -520,6 +520,8 @@ void nx_start(void)
 
   /* Initialize RTOS Data ***************************************************/
 
+  drivers_early_initialize();
```

Comment:
I'm curious how you ensure that all early initialize can be executed before idle tcb initialized.

Reply:
I'm not sure if I understand correctly. Currently `drivers_early_initialize` calls `notesnap_register` etc. which do not rely on idle tcb intialization. What's you concern here?

Reply:
Are you sure you have run the current trace function? There is a potential issue here

```
nx_start
|
 ->sched_trace_begin
   |
    ->trace_begin
      |
       ->sched_note_begin
         |
          ->sched_note_event
            |
             ->sched_note_event_ip
               |
                ->this_task()
```
                
this_task will be NULL and unable to record the tcb pid, cpu, and priority correctly. how abort move `sched_trace_begin() ` after idle initialization?